src/map/Pins.js

```diff
@@ -31,28 +51,48 @@
             .style("opacity", this.opacity())
         ;
 
-        this.pinsPaths = this._pinsTransform.selectAll(".data").data(this.visible() ? this.data() : [], function (d) { return d[0]; });
+        this.pinsPaths = this._pinsTransform.selectAll(".pin").data(this.visible() ? this.data() : [], function (d) { return d[0]; });
         var context = this;
-        this.pinsPaths.enter().append("path")
+        var gPinEnter = this.pinsPaths.enter().append("g").classed("pin",true);
+        gPinEnter
+            .append("path")
             .attr("class", "data")
-            .attr("d", function (d) {
-                return "M 0,0 C -2,-20 -10,-22 -10,-30 A 10,10 0 1,1 10,-30 C 10,-22 2,-20 0,0 z M -2,-30";
-            })
             .call(this._selection.enter.bind(this._selection))
             .on("click", function (d) {
                 context.click(context.rowToObj(d), "weight", context._selection.selected(this));
             })
             .append("title")
         ;
-        this.pinsPaths
-            .attr("stroke-width", 1.5 + "px")
+        gPinEnter
+            .append("text")
+            .attr("text-anchor","middle")
+        ;
+        this.pinsPaths.selectAll("text")
+            .style("stroke", this.fontColor())
+            .style("fill", this.fontColor())
+            .style("font-size", this.fontSize())
+            .style("font-family", this.fontFamily())
+            .style("dominant-baseline",this.textBaseline())
+            .attr("dx",0)
+            .attr("dy",context.pinTextDY())
+            .text(function(d){
+                return d[2] && d[2].text ? d[2].text : "";
+            });
+        this.pinsPaths.selectAll("path.data")
+            .attr("d", function (d) {
+                return context.svgPinPath();
+            })
```

Comment:
The following would suffice:
```
.attr("d", this.svgPinPath())
```

And rather than calling that function for each pin - you could call it once and store it in a var and then use that for each pin.